### DISCUSSION TITLE
Single Block for H1, H2, H3 etc

### DISCUSSION THREAD

**Discussion by dm-siegel** [GrapesJS/grapesjs#5143] [Q&A] [answered]
I would like to add a block for headings that allows the user to select the actual tag (h1, h2...).  Is it possible to create a component that changes it's own tag on property or attribute change?  Basically drop a "heading" component on the canvas and then allow the user to select h1, h2, etc from a selection attribute/property.  Once selected, it will actually use the selected tag.

I have looked into creating a custom tag <heading> tag with an attribute <heading tagType="h2">, then using the view to render the actual h2 tag.  It gets complicated when I am wanting to do a getHtml() or if I want to apply style/classes.

**Comment by artf** [comment, marked as answer]
Check this plugin https://github.com/olivmonnier/grapesjs-plugin-header it uses the old API but the concept for traits it's the same

**Reply by dm-siegel** [reply]
This was it... i should be better about looking through the existing github plugins.  Thanks!
